File: 202. Happy Number/Solution.py
class Solution:
    def isHappy(self, n: int) -> bool:
        s = set()
        while n!=1 and not (n in s):
            s.add(n)
            summ = 0
            while n!=0:
                d = n%10
                summ += d*d
                n //= 10
            n = summ
        if n==1:
            return True
        else:
            return False

File: 202. Happy Number/Solution.cpp
class Solution {
public:
    bool isHappy(int n) {
        unordered_set<int> s;
        while(n!=1 && !s.count(n)){
            s.insert(n);
            long long sum = 0;
            while(n != 0){
                int d = n%10;
                sum += d*d;
                n /= 10;
            }
            n = sum;
        }
        if(n==1){
            return true;
        }else{
            return false;
        }
    }
};
